src/audio/google/google_rtc_audio_processing_mock.c

```diff
@@ -145,11 +147,17 @@ int GoogleRtcAudioProcessingProcessCapture_int16(GoogleRtcAudioProcessingState *
 	int16_t *ref = state->aec_reference;
 	int16_t *mic = (int16_t *) src;
 	int16_t *out = dest;
-	int n;
+	int channels;
+	int n, m;
+
+	channels = MIN(state->num_capture_channels, state->num_aec_reference_channels);
+	channels = MIN(state->num_output_channels, channels);
 
 	memset(dest, 0, sizeof(int16_t) * state->num_output_channels * state->num_frames);
 	for (n = 0; n < state->num_frames; ++n) {
-		*out = *mic + *ref;
+		for (m = 0; m < channels; m++)
```

Comment:
is it a bug? since previous code does not have channels loop.

Reply:
The original code mixed only to first channel, with this loop we can mix to other output channels. But according to Ranjani's comment we should decide what to do with non-matching channels counts.